source/fundamentals/typescript.txt

```diff
@@ -104,22 +108,30 @@ genus and color of dogs and cats:
      classification: { genus: "Canis" | "Felis"; color: string };
    }
 
-The following code snippet correctly raises a type error when specifying
-the genus of an unsupported animal in a query:
+The driver does not raise a type error for the following code sample,
+even though the value of ``classification.color`` is a boolean
+instead of a string:
 
 .. code-block:: typescript
 
-   database
-     .collection<ClassificationPet>("<your collection>")
-     .find({ "classification.genus": "Sylvilagus" });
+   await collection.findOneAndDelete({ "classification.color": false })
+
+You can enable type-checking by constructing filters as ``StrictFilter`` types.
 
-The type error raised by the preceding code snippet is as follows:
+.. warning::
+   
+   Use of the ``StrictFilter`` type to create filters is experimental and currently has
+   unknown performance implications.
```

Comment:
```suggestion
   Use of the ``StrictFilter`` type to create filters is experimental and may have type
   errors in places where there should be none.
```

We aren't worried about the performance, it is more about the usability hurdles that can get in the way of writing a "correct query" from the server's perspective but that our typescript type can't handle.